### PR TITLE
CDAP-16994 fix schema equality check during pipeline deployment

### PR DESCRIPTION
**Pull request by albertshau**
Fixed a bug that would cause pipeline deployment to fails if
two stages have the same output stage and their output schemas
are exactly the same except for the schema name.

Implemented a schema equality check that ignores record name
when checking equality, and used it for the validation checks.

### CODE REVIEW

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-DUT7255

File: cdap-api-common/src/main/java/io/cdap/cdap/api/data/schema/SchemaHash.java [hunk 1]
```diff
@@ -41,14 +41,19 @@
...
+  public SchemaHash(Schema schema, boolean includeRecordName) {
+    hash = computeHash(schema, includeRecordName);
   }
 
   /**
    * Creates a SchemaHash by reading the hash value from the given {@link ByteBuffer}.
    * The position of the {@link ByteBuffer} will be moved to the byte after the hash
    * value.
    */
+  @Deprecated
```
Comment (chtyim) [inline cdap-api-common/src/main/java/io/cdap/cdap/api/data/schema/SchemaHash.java]: Why this is deprecated?

**Review by chtyim — COMMENTED** [review]
Just one comment

File: cdap-api-common/src/main/java/io/cdap/cdap/api/data/schema/SchemaHash.java (hunk 1 above)
Comment (albertshau) [reply cdap-api-common/src/main/java/io/cdap/cdap/api/data/schema/SchemaHash.java]: was a intermediate thing i forgot to remove, will remove

**Review by chtyim — APPROVED** [review]

**Pull request merged**